### ISSUE TITLE
postcss-custom-media (9.0.0)

### ISSUE THREAD

**Issue by nileshprajapati** [csstools/postcss-plugins#705]
### Bug description

The recently updated postcss plugin version 9 and have noticed unusual CSS output compared to version 8.0.2.

Issue 1:- Double media queries nested within.
Issue 2:- output contains `@media screen and (color: ...  )` & `@media screen and (max-color: ... )`.

I have reverted the plugin to v8.0.2 and confirmed that the plugin works as expected.

### Source CSS

```css
@custom-media --screen-md-n-above (width >= 48em);
@custom-media --screen-lg-n-above (width >= 64em);
@custom-media --screen-xl-n-above (width >= 80em);

:where(h1, h2, h3, h4, h5, h6) {
	font-size: 20px);
	font-weight: 600;
	line-height: 1.5;

	@media screen and (--screen-md-n-above) {
		font-size: 22px;
	}

	@media screen and (--screen-lg-n-above) {
		font-size: 26px;
	}

	@media screen and (--screen-xl-n-above) {
		font-size: 28px
	}
}
```

### Expected CSS

```css
:where(h1, h2, h3, h4, h5, h6){
	font-size: 20px;
	font-weight: 600;
	line-height:1.5;
}
@media screen and (min-width: 48em){
:where(h1, h2, h3, h4, h5, h6){
		font-size: 22px;
}
	}
@media screen and (min-width: 64em){
:where(h1, h2, h3, h4, h5, h6){
		font-size: 26px;
}
	}
@media screen and (min-width: 80em){
:where(h1, h2, h3, h4, h5, h6){
		font-size: 28px;
}
	}
```

### Actual CSS

```css
:where(h1, h2, h3, h4, h5, h6) {
	font-size: 20px;
	font-weight: 600;
	line-height: 1.5;
}
@media (min-width: 48em) {
	@media screen and (max-color: 2147477350) {
		:where(h1, h2, h3, h4, h5, h6) {
			font-size: 22px;
		}
	}
}
@media not all and (min-width: 48em) {
	@media screen and (color: 2147477350) {
		:where(h1, h2, h3, h4, h5, h6) {
			font-size: 22px;
		}
	}
}
@media (min-width: 64em) {
	@media screen and (max-color: 2147477350) {
		:where(h1, h2, h3, h4, h5, h6) {
			font-size: 26px;
		}
	}
}
@media not all and (min-width: 64em) {
	@media screen and (color: 2147477350) {
		:where(h1, h2, h3, h4, h5, h6) {
			font-size:26px;
		}
	}
}
@media (min-width: 80em) {
	@media screen and (max-color: 2147477350) {
		:where(h1, h2, h3, h4, h5, h6) {
			font-size: 28px;
		}
	}
}
@media not all and (min-width: 80em) {
	@media screen and (color: 2147477350) {
		:where(h1, h2, h3, h4, h5, h6) {
			font-size: 28px;
		}
	}
}
```

### Does it happen with `npx @csstools/csstools-cli <plugin-name> minimal-example.css`?

N/A

### Extra config

List of plugins currently in use:
"autoprefixer": "^10.4.13",
"cssnano": "^5.1.14",
"postcss": "^8.4.19",
"postcss-cli": "^10.0.0",
"postcss-custom-media": "^8.0.2",
"postcss-discard-comments": "^5.1.2",
"postcss-discard-empty": "^5.1.1",
"postcss-import": "^15.0.0",
"postcss-media-minmax": "^5.0.0",
"postcss-nesting": "^10.2.0"

### What plugin are you experiencing this issue on?

PostCSS Custom Media Queries

### Plugin version

9

### Node Version

v19.0.0

### Validations

- [X] Follow our [Code of Conduct](https://github.com/csstools/postcss-plugins/blob/master/CODE_OF_CONDUCT.md)
- [X] Check that there isn't already an issue that request the same feature to avoid creating a duplicate.

### Would you like to open a PR for this bug?

- [X] I'm willing to open a PR

**Comment by romainmenke** [comment]
Hi @nileshprajapati, thank you for reaching out about this!

That output is actually expected and documented here : https://www.npmjs.com/package/postcss-custom-media#logical-evaluation-of-complex-media-queries

We might be able to had a few more heuristics to simplify the output.
You case might be possible to simplify.

Was there anything that didn't work in a browser?
Or did you only not expect to see this generated CSS?

**Comment by nileshprajapati** [comment]
HI @romainmenke 

Thank you for checking this for me.

**Was there anything that didn't work in a browser?**
No, everything works as expected.

**Or did you only not expect to see this generated CSS?**
I only noticed this once I had the plugin working again.

At first updating to v9, the plugin wasn't outputting the custom media variable value.

Thanks,
Nilesh

**Comment by romainmenke** [comment]
> I only noticed this once I had the plugin working again.
> At first updating to v9, the plugin wasn't outputting the custom media variable value.

Can you share what went wrong?
That sounds like an issue (maybe a different one)

**Comment by nileshprajapati** [comment]
When I updated to v9, it was output like the "Source CSS" example above that included @custom-media and the custom-media variable.

**Comment by romainmenke** [comment]
What did you change to resolve that issue?

**Comment by nileshprajapati** [comment]
Copy an existing project node-module, postcss.config.js, package folder over and upgrade each plugin one by one.
But also I did a clean node-module install as well.

**Comment by romainmenke** [comment]
Got it, thank you for providing this info.
We will look into your original request in the next few days.

**Comment by romainmenke** [comment]
@nileshprajapati I am going to keep this open for a bit longer until we can get an update out :)

**Comment by romainmenke** [comment]
@nileshprajapati A fix was just released as `v9.0.1`
https://www.npmjs.com/package/postcss-custom-media

Thank you again for reporting!
